test_scripts/Resumption/Handling_errors_from_HMI/054_resume_IVD_subscription_after_erroneous_response_all_moduleTypes.lua

```diff
@@ -0,0 +1,93 @@
+-- 2. Unexpected disconnect and Reconnect are performed
+-- 3. App re-registers with actual HashId
+-- 4. HMI responds with erroneous resultCode for one interior vehicle data type
+-- SDL does:
+--  - process unsuccess response from HMI
+--  - remove already restored data
+--  - respond RegisterAppInterfaceResponse(success=true,result_code=RESUME_FAILED) to mobile application
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Resumption/Handling_errors_from_HMI/commonResumptionErrorHandling')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+-- [[ Local Function ]]
+local function getExpectedData()
+  local expectedData = {}
+  for _, moduleTypeValue in pairs(common.rcModuleTypes) do
+    table.insert(expectedData, common.geInteriorVDvalue(moduleTypeValue, true))
+    table.insert(expectedData, common.geInteriorVDvalue(moduleTypeValue, false))
+  end
+  return expectedData
+end
+
+local function checkResumptionData()
+  local actualData = {}
+  local expectedData = getExpectedData()
+  local expectedNumber = 2 * #common.rcModuleTypes - 1 --`-1` because SDL does not send revert request for data
+                                                       -- with erroneous response
+  common.getHMIConnection():ExpectRequest("RC.GetInteriorVehicleData")
+  :Do(function(exp, data)
+      common.log("Received " .. data.method .. ", moduleType: " .. data.params.moduleType .. ", subscribe: " ..
+        tostring(data.params.subscribe))
+      if exp.occurences == #common.rcModuleTypes then
+        for k = #expectedData, 1, -1 do
+          if expectedData[k].moduleType == data.params.moduleType and expectedData[k].subscribe == false then
+            table.remove(expectedData, k)
+          end
+        end
+        local function sendResponse()
+          common.log("Sent " .. data.method .. ": GENERIC_ERROR, moduleType: " .. data.params.moduleType ..
+            ", subscribe: " .. tostring(data.params.subscribe))
+          common.getHMIConnection():SendError(data.id, data.method, "GENERIC_ERROR", "info message")
+        end
+        common.run.runAfter(sendResponse, 300)
+      else
+        common.log("Sent " .. data.method .. ": SUCCESS, moduleType: " .. data.params.moduleType .. ", subscribe: " ..
+          tostring(data.params.subscribe))
+        local resParams = { }
+        resParams.moduleData = common.getActualModuleIVData(data.params.moduleType, data.params.moduleId)
+        resParams.isSubscribed = data.params.subscribe
+        common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", resParams)
+      end
+    end)
+  :ValidIf(function(exp, data)
+      table.insert(actualData, data.params)
+      return common.interiorVDvalidation(exp.occurences, expectedNumber, actualData, expectedData)
```

Comment:
```suggestion
      if exp.occurences == expectedNumber then
        return common.validateInteriorVD(actualData, expectedData)
      end
```
same applies everywhere else this function is used

Reply:
@jacobkeeler Done (see 1012029)